### ISSUE TITLE
check failed: exit status 22

### ISSUE THREAD

**Issue by elgohr** [nulldriver/maven-resource#23]
When the resource is used without a former put, e.g.:
`
resources:
  - name: snapshot
    type: maven-resource
    source:
      url: *__ARTIFACTORY_RELEASE_URL__
      snapshot_url: *__ARTIFACTORY_SNAPSHOT_URL__
      artifact: *__ARTIFACTORY_ARTIFCAT___

jobs:
  - name: job
    plan:
      - get: snapshot
        trigger: true
      - task: do-whatever
...

`
And the `url` and `release_url` are filled, like shown above, the resource uses the release version.
Should this probably be documented?

When there is no release version it fails to get the version via curl.
This results in the following error
![screenshot from 2018-05-01 14-08-13](https://user-images.githubusercontent.com/19632501/39650914-374eae5c-4fea-11e8-925d-7bf5ecd4519d.png)

The error message in check (line 76) is not shown and debugging is not that easy.

**Comment by patrickcrocker** [comment]
I've fixed the error reporting (`1.3.7-rc.8`) so that it is easier to understand what failed.  Now you should see something like this:
```
[ERROR] curl: (22) The requested URL returned error: 404
[ERROR] failed to download maven-metadata.xml from: https://nexus.nulldriver.com/repository/maven-releases/com/nulldriver/does-not-exist/maven-metadata.xml
```

The statement about line 31 not working for Concourse is not correct.  Concourse passes in the last known version of the resource as `.version.version`.  I do acknowledge that this value will be empty for the case that the resource does not exist yet.  How the `maven-resource` handles this is now up for debate.  :-)

At this point I'm second guessing my decision to support both `url` and `snapshot_url`...  It seems that if you need to make that level of distinction in your pipeline, you should probably have two defined resources, one for snapshot and one for releases...  thoughts?

**Comment by designed4device** [comment]
Agreed, I prefer the snapshots and releases to be separate resources in my pipelines for clarity. In the resources current state, it appears to be a necessity for the use case of a pipeline that is only using the resource for gets in order to retrieve a snapshot.
I think supporting a single url param would reduce some complexity in the codebase and make the resource's functionality a bit easier to understand.

**Comment by designed4device** [comment]
@patrickcrocker I ended up doing an overhaul in my fork that included removing the snapshot_url param. Have you thought any further on removing it for this repo?
